.github/workflows/test-utils.yml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019-2024 Tauri Programme within The Commons Conservancy
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-License-Identifier: MIT
+
+name: test utils
+
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
+    paths:
+      - '.github/workflows/test-utils.yml'
+      - 'crates/tauri-utils/**'
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_PROFILE_DEV_DEBUG: 0 # This would add unnecessary bloat to the target folder, decreasing cache efficiency.
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ${{ matrix.platform.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - {
+              target: x86_64-pc-windows-msvc,
+              os: windows-latest,
+              toolchain: '1.77.2',
+              cross: false,
+              command: 'test'
+            }
+          - {
+              target: x86_64-unknown-linux-gnu,
+              os: ubuntu-latest,
+              toolchain: '1.77.2',
+              cross: false,
+              command: 'test'
+            }
+          - {
+              target: aarch64-apple-darwin,
+              os: macos-14,
+              toolchain: '1.77.2',
+              cross: false,
+              command: 'test'
+            }
+          - {
+              target: aarch64-apple-ios,
+              os: macos-latest,
+              toolchain: '1.77.2',
+              cross: false,
+              command: 'build'
+            }
+          - {
+              target: aarch64-linux-android,
+              os: ubuntu-latest,
+              toolchain: '1.77.2',
+              cross: true,
+              command: 'build'
+            }
+        features:
+          - { args: --no-default-features, key: no-default }
+          - { args: --all-features, key: all }
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.platform.toolchain }}
+          targets: ${{ matrix.platform.target }}
+
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: v3
+          save-if: ${{ matrix.features.key == 'all' }}
+
+      - name: test
+        if: ${{ !matrix.platform.cross }}
+        # Using --lib --bins --tests to skip doc tests
+        run: cargo ${{ matrix.platform.command }} --target ${{ matrix.platform.target }} ${{ matrix.features.args }} --lib --bins --tests --manifest-path crates/tauri-utils/Cargo.toml
+
+      - name: test (using cross)
+        if: ${{ matrix.platform.cross }}
+        run: |
+          cargo install cross --git https://github.com/cross-rs/cross --rev 51f46f296253d8122c927c5bb933e3c4f27cc317 --locked
+          cross ${{ matrix.platform.command }} --target ${{ matrix.platform.target }} ${{ matrix.features.args }} --lib --bins --tests --manifest-path crates/tauri-utils/Cargo.toml
```

Comment:
can't we merge this into test-core.yml with a manifest-path option in the matrix?

Reply:
> can't we have a new step or two above the 2 you modified that builds tauri-utils?

brain didn't work 😂

Reply:
at least you have a brain, sometimes i feel like my hands write on their own somehow